### PR TITLE
adding new rules based on private Linux sample(s)

### PR DESCRIPTION
**Pull request by mike-hunhoff**

### CODE REVIEW

File: nursery/create-detached-thread-on-linux.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+rule:
+  meta:
+    name: create detached thread on Linux
```
Comment (mr-tz) [inline nursery/create-detached-thread-on-linux.yml:3]:
```suggestion
    name: detach thread on Linux
```

Comment (mr-tz) [reply nursery/create-detached-thread-on-linux.yml:3]: or `mark thread detached`

File: nursery/get-user-database-entry-on-linux.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    namespace: host-interaction/session
+    authors:
+      - michael.hunhoff@mandiant.com
+    scope: basic block
+  features:
+    - and:
+      - os: linux
+      - or:
+        - api: getpwuid
+        - api: getpwuid_r
```
Comment (mr-tz) [inline nursery/get-user-database-entry-on-linux.yml:13]: also `getpwnam` and `getpwnam_r`?

File: nursery/get-user-database-entry-on-linux.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+rule:
+  meta:
+    name: get user database entry on Linux
```
Comment (mr-tz) [inline nursery/get-user-database-entry-on-linux.yml:3]:
```suggestion
    name: get password database entry on Linux
```

File: nursery/communicate-with-kernel-module-via-netlink-socket-on-linux.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+rule:
+  meta:
+    name: communicate with kernel module via Netlink socket on Linux
+    namespace: host-interaction/kernel
```
Comment (mr-tz) [inline nursery/communicate-with-kernel-module-via-netlink-socket-on-linux.yml:4]: not sure the namespace is great, but cannot think of anything else right now

**Review by mr-tz — COMMENTED** [review]
nice, work, a few suggestions inline

File: nursery/communicate-with-kernel-module-via-netlink-socket-on-linux.yml (hunk 4 above)
Comment (mike-hunhoff) [reply nursery/communicate-with-kernel-module-via-netlink-socket-on-linux.yml:4]: Agreed -  I had the same thoughts when creating the rule. I'll leave it as-is for now.

File: nursery/get-user-database-entry-on-linux.yml (hunk 3 above)
Comment (mike-hunhoff) [reply nursery/get-user-database-entry-on-linux.yml:3]: Nice call

File: nursery/get-user-database-entry-on-linux.yml (hunk 2 above)
Comment (mike-hunhoff) [reply nursery/get-user-database-entry-on-linux.yml:13]: Nice catch

**Review by mr-tz — APPROVED** [review]
great, thank you!

**Pull request merged**